### DISCUSSION TITLE
OIDC_id_token is not added when OIDCSessionType client-cookie

### DISCUSSION THREAD

**Discussion by trussrod75** [OpenIDC/mod_auth_openidc#650] [Q&A] [answered]
Hi, just an observation and I guess looking for confirmation.  When I set OIDCSessionType client-cookie and OIDCPassIDTokenAs claims serialized (or likely OIDCPassIDTokenAs serialized for that matter). the OIDC_id_token header or var is not added.  As soon as a change OIDCSessionType to server-cache (or comment it out completely) the OIDC_id_token is added.  Can you confirm this is expected?

#OIDCSessionType               client-cookie
OIDCPassClaimsAs              both
OIDCPassIDTokenAs             claims serialized

This passes OIDC_id_token.  As soon as I uncomment  OIDCSessionType and restart OIDC_id_token is no longer passed.

I haven't decided on OIDCSessionType or my clustering / session / cache details.  I think to to have a robust cluster I will likely need to use server-cache anyway, there is more research to be done by me on that for sure.  I could find anywhere that OIDC_id_token requires server-cache so I thought I would raise it to see if it's expected.

Many thanks!
Ryan

**Comment by zandbelt** [comment, marked as answer]
this is expected behaviour: storing the`id_token` in the client side cookie would almost double its size and it duplicate other information already parsed into the session (e.g. claims); I've added a note here: https://github.com/zmartzone/mod_auth_openidc/commit/d071a4af137e816f7f31b5e0339e9d4f0289d703

**Reply by trussrod75** [reply]
Much appreciated, thanks!
